service/src/main/java/bio/terra/workspace/app/controller/ControlledGcpResourceApiController.java

```diff
@@ -85,7 +83,11 @@ public ResponseEntity<ApiCreatedControlledGcpGcsBucket> createBucket(
     Workspace workspace =
         workspaceService.validateMcWorkspaceAndAction(
             userRequest, workspaceUuid, getSamAction(body.getCommon()));
-    String resourceLocation = getResourceLocation(workspace, body.getGcsBucket().getLocation());
+    // Buckets are regional resources.
+    String resourceLocation =
+        GcpUtils.parseRegion(
```

Comment:
Instead of calling `parseRegion` everywhere, how about making an alternate endpoint in GcpCloudContextService: `getResourceRegion` that would validate the incoming location - or use the cloud context default - and parse out the region component?